### ISSUE TITLE
Erroneous `not-an-iterable` when inheriting from `NamedTuple`

### ISSUE THREAD

**Issue by One-Nose** [pylint-dev/pylint#9538]
### Bug description

```python
# pylint: disable=missing-module-docstring
# pylint: disable=missing-class-docstring
# pylint: disable=missing-function-docstring

from __future__ import annotations

from typing import NamedTuple


class Vec3D(NamedTuple):
    x: int
    y: int
    z: int

    def inc(self) -> Vec3D:
        return Vec3D(a + 1 for a in self)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:16:36: E1133: Non-iterable value self is used in an iterating context (not-an-iterable)
```

### Expected behavior

Should not produce error.

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)]
```

**Comment by One-Nose** [comment]
This was already reported in #2568 as a problem with all typing generics and was closed because it couldn't be reproduces. However, I encountered this error specifically with `NamedTuple` (as also described in https://github.com/pylint-dev/pylint/issues/2568#issuecomment-461906170).

**Comment by One-Nose** [comment]
Update: This appears to be a problem with all of `NamedTuple`'s attributes, including 'self._replace`, `self._asdict`, etc.

**Comment by One-Nose** [comment]
Nevermind, appears to be a duplicate of #7891.
